script.go

```diff
@@ -0,0 +1,682 @@
+
+	var cols []string
+	if len(*columns) > 0 {
+		cols = strings.Split(*columns, ",")
+	}
+
+	if err := flags.Parse(args); err != nil {
+		return nil, fmt.Errorf("%w: %s", errUsage, err)
+	}
+	args = flags.Args()
+	if len(args) < 2 {
+		return nil, fmt.Errorf("%w: expected table and key", errUsage)
+	}
+
+	return func(*script.State) (stdout, stderr string, err error) {
+		tbl, txn, err := getTable(db, args[0])
+		if err != nil {
+			return "", "", err
+		}
+
+		var buf strings.Builder
+		var w io.Writer
+		if *file == "" {
+			w = &buf
+		} else {
+			f, err := os.OpenFile(s.Path(*file), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0644)
+			if err != nil {
+				return "", "", fmt.Errorf("OpenFile(%s): %s", *file, err)
+			}
+			defer f.Close()
+			w = f
+		}
+
+		var it iter.Seq2[any, uint64]
+		switch query {
+		case queryCmdList:
+			it, err = tbl.List(txn, *index, args[1])
+		case queryCmdLowerBound:
+			it, err = tbl.LowerBound(txn, *index, args[1])
+		case queryCmdPrefix:
+			it, err = tbl.Prefix(txn, *index, args[1])
+		case queryCmdGet:
+			it, err = tbl.List(txn, *index, args[1])
+			if err == nil {
+				it = firstOfSeq2(it)
+			}
+		default:
+			panic("unknown query enum")
+		}
+		if err != nil {
+			return "", "", fmt.Errorf("query: %w", err)
+		}
+
+		err = writeObjects(tbl, it, w, cols, *format)
+		if err != nil {
+			return "", "", err
+		}
+
+		if *delete {
+			wtxn := db.WriteTxn(tbl.Meta)
+			count := 0
+			for obj := range it {
+				_, hadOld, err := tbl.Delete(wtxn, obj)
+				if err != nil {
+					wtxn.Abort()
+					return "", "", err
+				}
+				if hadOld {
+					count++
+				}
+			}
+			s.Logf("Deleted %d objects\n", count)
+			wtxn.Commit()
+		}
+
+		return buf.String(), "", err
+	}, nil
+}
+
+func firstOfSeq2[A, B any](it iter.Seq2[A, B]) iter.Seq2[A, B] {
+	return func(yield func(a A, b B) bool) {
+		for a, b := range it {
+			yield(a, b)
+			break
+		}
+	}
+}
+
+func writeObjects(tbl *AnyTable, it iter.Seq2[any, Revision], w io.Writer, columns []string, format string) error {
+	if len(columns) > 0 && format != "table" {
+		return fmt.Errorf("-columns not supported with non-table formats")
+	}
+	switch format {
+	case "yaml":
+		sep := []byte("---\n")
+		first := true
+		for obj := range it {
+			if !first {
+				w.Write(sep)
+			}
+			first = false
+
+			out, err := yaml.Marshal(obj)
+			if err != nil {
+				return fmt.Errorf("yaml.Marshal: %w", err)
+			}
+			if _, err := w.Write(out); err != nil {
+				return err
+			}
+		}
+		return nil
+	case "json":
+		sep := []byte("\n")
+		first := true
+		for obj := range it {
+			if !first {
+				w.Write(sep)
+			}
+			first = false
+
+			out, err := json.Marshal(obj)
+			if err != nil {
+				return fmt.Errorf("json.Marshal: %w", err)
+			}
+			if _, err := w.Write(out); err != nil {
+				return err
+			}
+		}
+		return nil
+	case "table":
+		header := tbl.TableHeader()
+		if header == nil {
+			return fmt.Errorf("objects in table %q not TableWritable", tbl.Meta.Name())
```

Comment:
Here's perhaps a good reason to change `NewTable[Obj any]` into `NewTable[Obj TableWritable]` (or maybe make a `TableObject` constraint). Again don't want to make this into a breaking change yet, but maybe for statedb v0.4...